source/extensions/filters/common/ext_authz/check_request_utils.cc

```diff
@@ -124,6 +124,10 @@ void CheckRequestUtils::setHttpRequest(
     std::string data(length, 0);
     buffer->copyOut(0, length, &data[0]);
     httpreq.set_body(std::move(data));
+
+    // Add in a header to detect when a partial body is used.
```

Comment:
We may also want to check that this header does not exist when the request is not a partial message. WDYT?

Reply:
Not exactly sure what you mean here. Only send the header if the body was actually truncated? Or only send the header if the `allow_partial_message` option is set...

The metadata header is currently only there if `max_request_bytes` is set. In order to detect if this feature is available, I needed to send it with a bool (`0|1`) value (vs only sending the header if the body is partial). This way, in the gRPC service, you can use the logic "if the header is there, then it is coming from an envoy with this feature (v1.11+) and the value indicates partial or not. In this case the work is done for you, otherwise this is an older envoy and you need to detect this on your own using C-L header or other means."

Now, if you mean only write the header if the config option `allow_partial_message: true`, then yeah, that could be done, but I did not want to add another parameter to `CheckRequestUtils::setHttpRequest` to expose a check for this. I did not think it was worth it as it meant it was harder to determine from the gRPC service if the new partial flag feature was available. That is, if the header is not sent when `allow_partial_message: false`, then the gRPC service cannot tell if it is envoy v1.10 or the partial feature is disabled in v1.11.

If you think that the metadata header may get in people's way when `allow_partial_message: false`, then I can expose the config option as a parameter to `CheckRequestUtils::setHttpRequest`. Unless maybe you see another way to get access to the config option without it as a parameter?

Reply:
What happens if the client request contains `x-envoy-auth-partial-body: 1` header? Not sure if we should trust the client.

Reply:
Good point. I'll skip that header during the copy.